tsdb/engine/tsm1/engine.go

```diff
@@ -41,19 +41,35 @@ const (
 
 // Statistics gathered by the engine.
 const (
-	statCacheCompactions              = "cacheCompactions"
-	statCacheCompactionError          = "cacheCompactionErr"
-	statCacheCompactionDuration       = "cacheCompactionDuration"
-	statTSMLevel1Compactions          = "tsmLevel1Compactions"
-	statTSMLevel1CompactionDuration   = "tsmLevel1CompactionDuration"
-	statTSMLevel2Compactions          = "tsmLevel2Compactions"
-	statTSMLevel2CompactionDuration   = "tsmLevel2CompactionDuration"
-	statTSMLevel3Compactions          = "tsmLevel3Compactions"
-	statTSMLevel3CompactionDuration   = "tsmLevel3CompactionDuration"
+	statCacheCompactions        = "cacheCompactions"
+	statCacheCompactionsActive  = "cacheCompactionsActive"
+	statCacheCompactionError    = "cacheCompactionErr"
+	statCacheCompactionDuration = "cacheCompactionDuration"
+
+	statTSMLevel1Compactions        = "tsmLevel1Compactions"
+	statTSMLevel1CompactionsActive  = "tsmLevel1CompactionsActive"
+	statTSMLevel1CompactionError    = "tsmLevel1CompactionErr"
+	statTSMLevel1CompactionDuration = "tsmLevel1CompactionDuration"
+
+	statTSMLevel2Compactions        = "tsmLevel2Compactions"
+	statTSMLevel2CompactionsActive  = "tsmLevel2CompactionsActive"
+	statTSMLevel2CompactionError    = "tsmLevel2CompactionErr"
+	statTSMLevel2CompactionDuration = "tsmLevel2CompactionDuration"
+
+	statTSMLevel3Compactions        = "tsmLevel3Compactions"
+	statTSMLevel3CompactionsActive  = "tsmLevel3CompactionsActive"
+	statTSMLevel3CompactionError    = "tsmLevel3CompactionErr"
+	statTSMLevel3CompactionDuration = "tsmLevel3CompactionDuration"
+
 	statTSMOptimizeCompactions        = "tsmOptimizeCompactions"
+	statTSMOptimizeCompactionsActive  = "tsmOptimizeCompactionsActive"
```

Comment:
Yea, that's what I was thinking. Separate counters for started vs finished. This could be fine by itself though.